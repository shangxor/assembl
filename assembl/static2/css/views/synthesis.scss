.synthesis-page {
  .affix {
    top: 15%;
  }

  .synthesis-side-menu {
    background-color: $white;
    padding: 30px;
    margin: 0 10px;

    .title-hyphen {
      display: block;
      width: 50px;
      margin: 10px 0 10px -35px;
    }

    .assembl-icon-down-open {
      font-size: $font-size-xxs;
      color: $first-color;
      margin: 0 10px 1px 0;
    }

    .link-container {
      display: flex;
      align-items: center;
      justify-content: space-between;
    }

    .active {
      background: rgba($first-color, 0.15);

      a {
        pointer-events: none;
      }
    }

    .side-menu-link-1 {
      overflow: auto;
      font-size: $font-size-s;
      color: $first-color;
      line-height: 22px;
      font-family: $font-family3;
      letter-spacing: 2px;
      padding: 4px 0;
    }

    .side-menu-link-2 {
      overflow: auto;
      margin-left: 20px;
      font-size: $font-size-xs;
      color: $first-color;
      line-height: 20px;
      font-family: $font-family3;
    }

    .side-menu-link-3 {
      overflow: auto;
      margin-left: 40px;
      font-size: $font-size-xs;
      color: $first-color;
      line-height: 20px;
      font-family: $font-family3;
    }

    .dark-title-4 {
      font-family: $font-family;
      font-size: $font-size-m;
    }
  }

  .synthesis-block .content-section {
    padding: 0 0 60px 0;
    text-align: justify;
    font-size: $font-size-m;
  }

  .synthesis-tree {
    padding-top: 60px;
  }

  .idea-synthesis-section {
    .content-section {
      padding: 0;
    }

    .title-section {
      margin: 0;
    }

    .section-title1 {
      font-size: 32px;
    }

    .section-title-2 {
      font-size: 26px;
    }

    .section-title-3 {
      font-size: 20px;
    }
  }
}

.create-synthesis-button {
  margin-bottom: 25px;
}

.create-synthesis-form {
  .admin-box {
    margin: 50px 0;
    padding: 40px;
    border-radius: 3px;
  }

  .action-button {
    margin-left: 20px;
  }
}
