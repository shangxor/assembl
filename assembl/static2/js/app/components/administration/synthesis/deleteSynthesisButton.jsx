// @flow
import React from 'react';
import { Button } from 'react-bootstrap';
import { type DocumentNode, graphql, type TVariables } from 'react-apollo';
import { Translate } from 'react-redux-i18n';

import { closeModal, displayModal } from '../../../utils/utilityManager';
import deleteSynthesisMutation from '../../../graphql/mutations/deleteSynthesis.graphql';

import DeletePostIcon from '../../common/icons/deletePostIcon/deletePostIcon';

type RefetchQuery = {
  query: DocumentNode,
  variables: TVariables
};

export type Props = {
  /** Class that is applied to the Link component  */
  linkClassName?: string,
  /** Post identifier */
  synthesisPostId: string,
  /** Array of refetch Queries */
  refetchQueries?: Array<RefetchQuery>,
  /** Modal custom message */
  modalBodyMessage?: string,
  /** callback function handled by the parent component */
  onDeleteCallback?: Function
};

type GraphQLProps = {
  /** Mutation function name issued with deletePostMutation */
  deleteSynthesis: Function
};

type LocalProps = Props & GraphQLProps;

const DeleteSynthesisButton = ({
  deleteSynthesis,
  linkClassName,
  synthesisPostId,
  refetchQueries,
  modalBodyMessage,
  onDeleteCallback
}: LocalProps) => {
  const displayConfirmationModal = () => {
    const title = <Translate value="debate.confirmDeletionTitle" />;
    const body = <Translate value={modalBodyMessage || 'debate.synthesis.confirmDeletionBody'} />;
    const footer = [
      <Button key="cancel" onClick={closeModal} className="button-cancel button-dark">
        <Translate value="debate.confirmDeletionButtonCancel" />
      </Button>,
      <Button
        key="delete"
        onClick={() => {
          deleteSynthesis({
            refetchQueries: refetchQueries || [],
            variables: { id: synthesisPostId }
          });
          closeModal();
          if (onDeleteCallback) {
            onDeleteCallback();
          }
        }}
        className="button-submit button-dark"
      >
        <Translate value="debate.confirmDeletionButtonDelete" />
      </Button>
    ];
    const includeFooter = true;
    return displayModal(title, body, includeFooter, footer);
  };
  return (
    <Button bsClass={linkClassName} onClick={displayConfirmationModal}>
      <DeletePostIcon />
    </Button>
  );
};

DeleteSynthesisButton.defaultProps = {
  deleteSynthesis: null,
  linkClassName: '',
  refetchQueries: [],
  modalBodyMessage: 'debate.confirmDeletionBody',
  onDeleteCallback: null
};

export default graphql(deleteSynthesisMutation, { name: 'deleteSynthesis' })(DeleteSynthesisButton);