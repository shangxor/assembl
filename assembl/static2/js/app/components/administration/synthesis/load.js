export const load = async () => {};
export const postLoadFormat = () => {};