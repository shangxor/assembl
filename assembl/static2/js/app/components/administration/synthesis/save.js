export const createMutationPromises = () => {};

export const save = () => {};