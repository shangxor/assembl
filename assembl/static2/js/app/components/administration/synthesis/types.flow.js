// @flow
import type { FileValue, I18nRichTextValue, I18nValue } from '../../form/types.flow';
import { PublicationStates } from '../../../constants';

export type SynthesisFormValues = {
  subject: I18nValue,
  body: I18nRichTextValue,
  image: FileValue,
};

export type MultilingualSynthesisPost = {
  // The ID of the object.
  id: string,
  publicationState: string,
  // Graphene Field modeling a relationship to a published synthesisPost.
  publishesSynthesis: {|
    // The ID of the object.
    id: string,
    // The type of Synthesis to be created
    synthesisType: SynthesisTypes,
    // A list of possible languages of the entity as LangStringEntry objects. The subject in various languages.
    subjectEntries: LangstringEntries,
    // A list of possible languages of the entity as LangStringEntry objects. The body in various languages.
    bodyEntries: LangstringEntries,
    // This is a header image document object that will be visible on the Synthesis view's header.A file metadata object, described by the Document object.
    img: FileValue
  |}
}
