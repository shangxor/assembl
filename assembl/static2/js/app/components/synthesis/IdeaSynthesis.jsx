// @flow

import * as React from 'react';
import { Translate } from 'react-redux-i18n';
import { Link } from 'react-router';

import StatisticsDoughnut from '../debate/common/statisticsDoughnut';
import PostsAndContributorsCount from '../common/postsAndContributorsCount';
import { getColumnInfos, getSentimentsCount, createDoughnutElements } from '../debate/common/announcement';
import { MESSAGE_VIEW } from '../../constants';
import type { SynthesisIdea } from './types.flow';

const SynthesisBody = ({ level, hasSiblings, value, stats }) => (
  <div className="synthesis-body" style={{ columnCount: !hasSiblings && level > 2 ? 2 : 'auto' }}>
    <p dangerouslySetInnerHTML={{ __html: value }} />
    {stats}
  </div>
);

const LinkToIdea = ({ href }) => (
  <Link className="idea-link" to={href}>
    {'> '}
    <Translate value="synthesis.seeConversation" />
  </Link>
);

const SynthesisStats = ({ numContributors, numPosts, ideaLink, posts, isMultiColumns, messageColumns }) => {
  const sentimentsCount = getSentimentsCount(posts);
  const columnInfos = getColumnInfos(messageColumns);
  const doughnutElements = isMultiColumns ? columnInfos : createDoughnutElements(sentimentsCount);
  return (
    <div className="synthesis-stats">
      <StatisticsDoughnut elements={doughnutElements} placement="after" />
      <PostsAndContributorsCount vertical numContributors={numContributors} numPosts={numPosts} showNumPosts={!isMultiColumns} />
      <LinkToIdea href={ideaLink} />
    </div>
  );
};

const SynthesisImage = ({ level, imgUrl, stats }) => {
  if (level === 1) {
    return (
      <div className="synthesis-image-container" style={imgUrl && { backgroundImage: `url(${imgUrl})` }}>
        {stats}
      </div>
    );
  }
  if (level === 2) {
    return (
      <div>
        <div className="image-container">
          <div className="synthesis-image-container" style={imgUrl && { backgroundImage: `url(${imgUrl})` }} />
        </div>
        {stats}
      </div>
    );
  }
  return null;
};

const IdeaSynthesis = (props: { idea: SynthesisIdea, hasSiblings: boolean, level: number, slug: string }) => {
  const { idea, hasSiblings, level, slug } = props;
  const { id, img, numContributors, numPosts, posts, messageViewOverride, messageColumns } = idea.live;
  const phaseIdentifier = 'thread'; // TODO: Proper phase identification
  // For now, syntheses can only have ideas from the "thread" phase.

  const imgUrl = img && img.externalUrl;
  const link = `/${slug}/debate/${phaseIdentifier}/theme/${id}`;
  const stats = (
    <SynthesisStats
      numContributors={numContributors}
      numPosts={numPosts}
      ideaLink={link}
      posts={posts}
      isMultiColumns={messageViewOverride === MESSAGE_VIEW.messageColumns}
      messageColumns={messageColumns}
    />
  );
  return (
    <div className={`${'idea-synthesis idea-synthesis-level-'}${level}`}>
      <SynthesisImage level={level} imgUrl={imgUrl} stats={stats} />
      <SynthesisBody value={idea.synthesisTitle} hasSiblings={hasSiblings} level={level} stats={level >= 3 ? stats : null} />
    </div>
  );
};

export default IdeaSynthesis;