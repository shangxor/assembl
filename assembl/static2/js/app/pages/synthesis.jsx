// @flow
import React from 'react';
import { connect } from 'react-redux';
import { compose, graphql } from 'react-apollo';
import { Col, Grid, Row } from 'react-bootstrap';
import debounce from 'lodash/debounce';

import { renderSynthesisBody } from '../utils/linkify';
import Header from '../components/common/header';
import Section from '../components/common/section';
import SynthesisQuery from '../graphql/SynthesisQuery.graphql';
import manageErrorAndLoading from '../components/common/manageErrorAndLoading';
import IdeaSynthesisTree from '../components/synthesis/IdeaSynthesisTree';
import { getChildren, getPartialTree } from '../utils/tree';
import SideMenu from '../components/synthesis/sideMenu';
import { getDomElementOffset } from '../utils/globalFunctions';
import type { Synthesis } from '../components/synthesis/types.flow';

type SynthesisProps = {
  synthesis: Synthesis,
  routeParams: { slug: string, synthesisId: string },
  synthesisPostId: string
};

type SynthesisState = {
  introBlock: ?HTMLElement,
  conclusionBlock: ?HTMLElement,
  bodyBlock: ?HTMLElement,
  sideMenuNode: ?HTMLElement,
  sideMenuIsHidden: boolean,
  ideaOnScroll?: string,
  sideMenuHeight: number,
  sideMenuOverflow: string,
  setSideMenuHeight: string
};

const sideMenuTopPercentage = 15;

export class DumbSynthesis extends React.Component<SynthesisProps, SynthesisState> {
  constructor(props: SynthesisProps) {
    super(props);
    this.state = {
      introBlock: null,
      conclusionBlock: null,
      bodyBlock: null,
      sideMenuNode: null,
      sideMenuIsHidden: true,
      sideMenuHeight: 0,
      sideMenuOverflow: 'auto',
      setSideMenuHeight: '100%'
    };
  }

  componentWillMount() {
    window.addEventListener('scroll', this.updateSideMenuVisibility);
  }

  componentWillUnmount() {
    window.removeEventListener('scroll', this.updateSideMenuVisibility);
  }

  updateIntroBlock = (node: HTMLElement) => {
    this.setState({ introBlock: node });
  };

  updateConclusionBlock = (node: HTMLElement) => {
    this.setState({ conclusionBlock: node });
  };

  updateBodyBlock = (node: HTMLElement) => {
    this.setState({ bodyBlock: node });
  };

  updateSideMenuNode = (node: HTMLElement) => {
    this.setState({ sideMenuNode: node });
  };

  updateSideMenuVisibility = debounce(() => {
    const { introBlock, conclusionBlock, sideMenuNode } = this.state;
    if (introBlock && conclusionBlock) {
      const introNodeTop = getDomElementOffset(introBlock).top;
      const introNodeHeight = introBlock.getBoundingClientRect().height;
      const sideMenuHeight = sideMenuNode && sideMenuNode.getBoundingClientRect().height;
      if (sideMenuHeight && sideMenuHeight > 0) {
        this.setState({ sideMenuHeight: sideMenuHeight });
      }
      const conclusionBlockTopOffset = getDomElementOffset(conclusionBlock).top;
      const firstIdeaTopOffset = introNodeTop + introNodeHeight;
      const scroll = window.pageYOffset;
      const { innerHeight, innerWidth } = window;
      const sideMenuTopOffset = innerHeight * sideMenuTopPercentage / 100;
      // this value is the gap between the top of sideMenu and the top of the window set by the css 'top' property
      if (sideMenuHeight && sideMenuHeight + sideMenuTopOffset > innerHeight) {
        this.setState({ sideMenuOverflow: 'scroll', setSideMenuHeight: '80%' });
      }

      const hasScrollReachedSynthesis = scroll + 60 > firstIdeaTopOffset;
      // 60 corresponds to the gap between the first Idea and the introduction block (synthesis-tree's padding)
      const isBottomReached = scroll + sideMenuTopOffset + this.state.sideMenuHeight + 90 >= conclusionBlockTopOffset;
      // 90 corresponds to the gap between the top of the conclusion block and the top of its div (padding)

      if (isBottomReached || !hasScrollReachedSynthesis || innerWidth < 1000) {
        this.setState({ sideMenuIsHidden: true });
      } else {
        this.setState({ sideMenuIsHidden: false });
      }
    }
  }, 100);

  handleScrollForMenu = (node: HTMLElement) => {
    this.setState({ ideaOnScroll: node.id });
  };

  render() {
    const { synthesis, routeParams, synthesisPostId } = this.props;
    const { sideMenuIsHidden, ideaOnScroll, sideMenuOverflow, setSideMenuHeight } = this.state;
    const { introduction, conclusion, body, ideas, subject } = synthesis;
    const sortedIdeas = [...ideas].sort((a, b) => {
      if (a.live.order < b.live.order) {
        return -1;
      }
      if (a.live.order > b.live.order) {
        return 1;
      }
      return 0;
    });
    const { roots, descendants } = getPartialTree(sortedIdeas);
    const hasSiblings = roots.length > 1;
    return (
      <div className="synthesis-page">
        <div className="background-light">
          <Header title={subject} imgUrl={synthesis.img ? synthesis.img.externalUrl : ''} type="synthesis" />
          <Grid fluid>
            {introduction && (
              <Section title="introduction" translate className="synthesis-block" innerRef={this.updateIntroBlock}>
                <Row>
                  <Col mdOffset={3} md={8} smOffset={1} sm={10}>
                    <div dangerouslySetInnerHTML={{ __html: introduction }} />
                  </Col>
                </Row>
              </Section>
            )}
            {body && (
              <Section title="" className="synthesis-block" innerRef={this.updateBodyBlock}>
                <Row>
                  <Col mdOffset={3} md={8} smOffset={1} sm={10}>
                    {renderSynthesisBody(body)}
                  </Col>
                </Row>
              </Section>
            )}
            <Row className="background-grey synthesis-tree">
              <Col
                md={3}
                className="affix"
                style={{
                  top: `${sideMenuTopPercentage} %`,
                  overflow: sideMenuOverflow,
                  height: setSideMenuHeight
                }}
              >
                <SideMenu
                  rootIdeas={roots}
                  descendants={descendants}
                  synthesisPostId={synthesisPostId}
                  ideaOnScroll={ideaOnScroll}
                  innerRef={this.updateSideMenuNode}
                  show={!sideMenuIsHidden}
                />
              </Col>
              <Col mdOffset={3} md={8} sm={11}>
                {roots.map((rootIdea, index) => (
                  <IdeaSynthesisTree
                    hasSiblings={hasSiblings}
                    key={rootIdea.id}
                    rootIdea={rootIdea}
                    index={index + 1}
                    parents={[]}
                    subIdeas={getChildren(rootIdea, descendants)}
                    slug={routeParams.slug}
                    handleScrollForMenu={this.handleScrollForMenu}
                  />
                ))}
              </Col>
            </Row>
            {conclusion && (
              <Section
                title="conclusion"
                translate
                className="synthesis-block"
                id="conclusion"
                innerRef={this.updateConclusionBlock}
              >
                <Row>
                  <Col mdOffset={3} md={8} smOffset={1} sm={10}>
                    <div dangerouslySetInnerHTML={{ __html: conclusion }} />
                  </Col>
                </Row>
              </Section>
            )}
          </Grid>
        </div>
      </div>
    );
  }
}

const mapStateToProps = state => ({
  lang: state.i18n.locale
});

export default compose(
  connect(mapStateToProps),
  graphql(SynthesisQuery, {
    options: props => ({
      variables: { id: props.params.synthesisId, lang: props.lang }
    }),
    props: ({ data }) => {
      if (data.error || data.loading) {
        return {
          error: data.error,
          loading: data.loading
        };
      }

      return {
        error: data.error,
        loading: data.loading,
        synthesis: data.synthesisPost.publishesSynthesis,
        synthesisPostId: data.synthesisPost.id
      };
    }
  }),
  manageErrorAndLoading({ displayLoader: true })
)(DumbSynthesis);