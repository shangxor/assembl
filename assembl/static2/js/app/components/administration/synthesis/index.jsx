// @flow
import React from 'react';
import { connect } from 'react-redux';
import { type ApolloClient, withApollo, compose } from 'react-apollo';
import { Row, Col } from 'react-bootstrap';
import { Field } from 'react-final-form';
import arrayMutators from 'final-form-arrays';
import { I18n } from 'react-redux-i18n';

// Components imports
import Section from '../../common/section';
import LoadSaveReinitializeForm from '../../form/LoadSaveReinitializeForm';
import Loader from '../../common/loader';
import AdminForm from '../../../components/form/adminForm';

import { createMutationPromises, save } from './save';
import { load } from './load';
import MultilingualTextFieldAdapter from '../../form/multilingualTextFieldAdapter';
import FileUploaderFieldAdapter from '../../form/fileUploaderFieldAdapter';
import Helper from '../../common/helper';

type Props = {
  client: ApolloClient,
  editLocale: string,
  lang: string
};

const loading = <Loader />;

const CreateSynthesisForm = ({ client, editLocale, lang }: Props) => (
  <LoadSaveReinitializeForm
    load={(fetchPolicy: FetchPolicy) => load(client, fetchPolicy, lang)}
    loading={loading}
    createMutationPromises={createMutationPromises(client, lang)}
    save={save}
    validate={() => {}}
    mutators={{ ...arrayMutators }}
    render={({ handleSubmit, pristine, submitting }) => (
      <div className="administration max-container create-synthesis-form">
        <AdminForm handleSubmit={handleSubmit} pristine={pristine} submitting={submitting}>
          <Row>
            <Col xs={12} md={10}>
              <div className="admin-box">
                <Section title="debate.syntheses.createNewSynthesis" translate>
                  <Field
                    editLocale={editLocale}
                    name="title"
                    component={MultilingualTextFieldAdapter}
                    label={I18n.t('debate.syntheses.title')}
                    required
                  />
                  <Field
                    editLocale={editLocale}
                    name="url"
                    component={MultilingualTextFieldAdapter}
                    label={I18n.t('debate.syntheses.url')}
                    required
                  />
                  <div className="flex">
                    <Field name="picture" component={FileUploaderFieldAdapter} label={I18n.t('debate.syntheses.picture')} />
                    <Helper helperText="Photo de l'encarté dans la page des synthèses" popOverClass=" " />
                  </div>
                </Section>
              </div>
            </Col>
          </Row>
        </AdminForm>
      </div>
    )}
  />
);

const mapStateToProps = ({ admin, i18n }) => ({
  editLocale: admin.editLocale,
  lang: i18n.locale
});

export default compose(connect(mapStateToProps), withApollo)(CreateSynthesisForm);