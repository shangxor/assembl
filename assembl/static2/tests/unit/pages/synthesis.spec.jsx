import React from 'react';
import ShallowRenderer from 'react-test-renderer/shallow';

import { DumbSynthesis } from '../../../js/app/pages/synthesis';

describe('Synthesis component', () => {
  it('should match Synthesis snapshot', () => {
    const props = {
      routeParams: { slug: 'slugId' },
      synthesis: {
        id: 'fooId',
        subject: 'Foo',
        img: { externalUrl: 'http://foo.com/foo' },
        body: '<p>yolo</p>',
        ideas: [
          {
            id: 'barId',
            title: 'Bar',
            img: { externalUrl: 'http://foo.com/bar' },
            synthesisTitle: 'Foo',
            numContributors: 0,
            numPosts: 0,
            posts: { edges: [] },
            phaseIdentifier: 'paseId',
            ancestors: []
          }
        ]
      },
      synthesisPostId: "yop"
    };
    const renderer = new ShallowRenderer();
    renderer.render(<DumbSynthesis {...props} />);
    const rendered = renderer.getRenderOutput();
    expect(rendered).toMatchSnapshot();
  });
});