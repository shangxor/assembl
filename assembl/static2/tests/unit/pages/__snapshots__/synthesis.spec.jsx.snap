// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Synthesis component should match Synthesis snapshot 1`] = `
<div
  className="synthesis-page"
>
  <div
    className="background-light"
  >
    <Connect(Header)
      imgUrl="http://foo.com/foo"
      title="Foo"
      type="synthesis"
    />
    <Grid
      bsClass="container"
      componentClass="div"
      fluid={true}
    >
      <Row
        bsClass="row"
        componentClass="div"
      >
        <Col
          bsClass="col"
          componentClass="div"
          md={8}
          mdOffset={3}
          sm={10}
          smOffset={1}
        >
          <p>
            yolo
          </p>
        </Col>
      </Row>
      <Row
        bsClass="row"
        className="background-grey synthesis-tree"
        componentClass="div"
      >
        <Col
          bsClass="col"
          className="affix"
          componentClass="div"
          md={3}
          style={
            Object {
              "height": "100%",
              "overflow": "auto",
              "top": "15 %",
            }
          }
        >
          <SideMenu
            descendants={Array []}
            ideaOnScroll={null}
            innerRef={[Function]}
            rootIdeas={
              Array [
                Object {
                  "ancestors": Array [],
                  "id": "barId",
                  "img": Object {
                    "externalUrl": "http://foo.com/bar",
                  },
                  "numContributors": 0,
                  "numPosts": 0,
                  "phaseIdentifier": "paseId",
                  "posts": Object {
                    "edges": Array [],
                  },
                  "synthesisTitle": "Foo",
                  "title": "Bar",
                },
              ]
            }
            show={false}
            synthesisPostId="yop"
          />
        </Col>
        <Col
          bsClass="col"
          componentClass="div"
          md={8}
          mdOffset={3}
          sm={11}
        >
          <IdeaSynthesisTree
            handleScrollForMenu={[Function]}
            hasSiblings={false}
            index={1}
            parents={Array []}
            rootIdea={
              Object {
                "ancestors": Array [],
                "id": "barId",
                "img": Object {
                  "externalUrl": "http://foo.com/bar",
                },
                "numContributors": 0,
                "numPosts": 0,
                "phaseIdentifier": "paseId",
                "posts": Object {
                  "edges": Array [],
                },
                "synthesisTitle": "Foo",
                "title": "Bar",
              }
            }
            slug="slugId"
            subIdeas={Array []}
          />
        </Col>
      </Row>
    </Grid>
  </div>
</div>
`;
